airbyte-local.sh

```diff
@@ -134,22 +79,39 @@ function writeSrcConfig() {
           ({}; . + {($a[2*$i]): ($a[2*$i + 1]|fromjson? // .)})' > "$tempdir/$src_config_filename"
 }
 
-# TODO: take catalog as input parameter
 function writeSrcCatalog() {
-    case $src_docker_image in
-        *pagerduty*)
-            echo $pagerduty_src_catalog | jq > $tempdir/$src_catalog_filename
-            stream_prefix=mypagerdutysrc__pagerduty__
-            ;;
-        *servicenow*)
-            echo $servicenow_src_catalog | jq > $tempdir/$src_catalog_filename
-            stream_prefix=myservicenowsrc__servicenow__
-            ;;
-        *)
-            echo "Error: $src_docker_image is currently not supported"
-            exit 1
-            ;;
-    esac
+    supported_src="pagerduty|servicenow|phabricator"
+    src_docker_image_regex="^.*airbyte-($supported_src)-source$"
+    if [[ "$src_docker_image" =~ $src_docker_image_regex ]]; then
+        source_type="$(sed 's/.*airbyte-\(.*\)-source/\1/' <<< "$src_docker_image")"
+        stream_prefix="my${source_type}src__${source_type}__"
+
+        if [ -s "$src_catalog_filepath" ]; then
+            echo "Using catalog: $src_catalog_filepath"
+            cat "$src_catalog_filepath" | jq > "$tempdir/$src_catalog_filename"
+        else
+            echo "Retrieving catalog from faros-ai/airbyte-connectors-configs"
+            http_response=$(curl -s --write-out "HTTPSTATUS:%{http_code}" \
+                            "https://raw.githubusercontent.com/faros-ai/airbyte-connectors-configs/main/catalogs/${source_type}.json")
```

Comment:
I propose we make this repo public. Also maybe we should start tagging versions in this repo so that we can lock down the version that is retrieved here.

Reply:
We shouldn't make that repo public, and my implementation will use the `discover` command from the source docker image to determine the catalog instead of relying on version-specific files.

Reply:
nice! sounds good